src/test/test-fileio.c

```diff
@@ -624,9 +619,9 @@ TEST(search_and_fopen) {
         r = unlink(name);
         assert_se(r == 0);
 
-        r = search_and_fopen(basename(name), "re", NULL, (const char**) dirs, &f, &p);
+        r = search_and_fopen(bn, "re", NULL, (const char**) dirs, &f, &p);
         assert_se(r == -ENOENT);
```

Comment:
When you touch the code, please use `ASSERT_ERROR(search_and_fopen(...), ENOENT);`